src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/AggregateCrossTargetProjectContextProvider.cs

```diff
@@ -241,6 +245,18 @@ private static string GetDisplayName(ConfiguredProject configuredProject, Projec
                 projectData.DisplayName;
         }
 
+        private Task ExecuteWithinLockAsync(Action action) => _gate.ExecuteWithinLockAsync(JoinableCollection, JoinableFactory, action);
+
+        private int _isInitialized;
+
+        private async Task EnsureInitialized()
+        {
+            if (Interlocked.CompareExchange(ref _isInitialized, 1, 0) == 0)
```

Comment:
Would `Interlocked.Exchange(ref _isInitialized, 1) == 0` be simpler?

Reply:
I ran a benchmark. `CompareExchange` is quite a bit faster.

| Method |        Mean |     Error |     StdDev | Median |
------------------------ |------------:|----------:|-----------:|-------:|
| Exchange | 158.0000 ns | 81.148 ns | 239.267 ns | 0.0 ns |
| CompareExchange_Match |  16.8293 ns | 16.653 ns |  44.162 ns | 0.0 ns |
| CompareExchange_NoMatch |   0.5814 ns |  1.393 ns |   3.790 ns | 0.0 ns |

```c#
[ClrJob]
public class ExchangeVsCompareExchangeBenchmarks
{
    private int _i;

    [IterationSetup]
    public void Setup() => _i = 0;

    [Benchmark]
    public void Exchange() => Interlocked.Exchange(ref _i, 1);

    [Benchmark]
    public void CompareExchange_Match() => Interlocked.CompareExchange(ref _i, 0, 1);

    [Benchmark]
    public void CompareExchange_NoMatch() => Interlocked.CompareExchange(ref _i, 1, 1);
}
```

```
BenchmarkDotNet=v0.11.1, OS=Windows 10.0.17763
Intel Xeon Silver 4110 CPU 2.10GHz, 2 CPU, 32 logical and 16 physical cores
  [Host] : .NET Framework 4.7.2 (CLR 4.0.30319.42000), 32bit LegacyJIT-v4.7.3190.0
  Clr    : .NET Framework 4.7.2 (CLR 4.0.30319.42000), 32bit LegacyJIT-v4.7.3190.0
```

Reply:
I am quite surprised by the results there actually. Let me dig a little more...

Reply:
Yeah those results above are garbage.

| Method |     Mean |     Error |    StdDev |   Median |
| ------------------------ |---------:|----------:|----------:|---------:|
| Exchange | 7.222 ns | 0.1440 ns | 0.4246 ns | 6.985 ns |
| CompareExchange_Match | 9.680 ns | 0.1886 ns | 0.2880 ns | 9.555 ns |
| CompareExchange_NoMatch | 7.586 ns | 0.0116 ns | 0.0109 ns | 7.587 ns |

```c#
[ClrJob]
public class ExchangeVsCompareExchangeBenchmarks
{
    private const int OperationCount = 100_000_000;

    private int _i;

    [Benchmark(OperationsPerInvoke = OperationCount * 2)]
    public void Exchange()
    {
        _i = 0;

        for (var i = 0; i < OperationCount; i++)
        {
            Interlocked.Exchange(ref _i, 0);
            Interlocked.Exchange(ref _i, 1);
        }
    }

    [Benchmark(OperationsPerInvoke = OperationCount * 2)]
    public void CompareExchange_Match()
    {
        _i = 0;

        for (var i = 0; i < OperationCount; i++)
        {
            Interlocked.CompareExchange(ref _i, 0, 1);
            Interlocked.CompareExchange(ref _i, 1, 0);
        }
    }

    [Benchmark(OperationsPerInvoke = OperationCount * 2)]
    public void CompareExchange_NoMatch()
    {
        _i = 0;

        for (var i = 0; i < OperationCount; i++)
        {
            Interlocked.CompareExchange(ref _i, 1, 1);
            Interlocked.CompareExchange(ref _i, 1, 1);
        }
    }
}
```

There's not much in it. I'd go for `Exchange` based on it being the simpler form, but it's not a biggie. My curiousity's been satisfied.